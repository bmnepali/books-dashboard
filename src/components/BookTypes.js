import React from 'react';
import PropTypes from 'prop-types';

const BookTypes = props => {
  return (
    <section>
      <h3 className="sidebar-heading">Book Types</h3>
      <ul className="nav flex-column mb-2">
        <li className="nav-item">
          <a href="#" className="nav-link" onClick={() => props.changeCategory('Biography')}>
            Biography
          </a>
        </li>
        <li className="nav-item">
          <a href="#" className="nav-link" onClick={() => props.changeCategory('Kids')}>
            Kids
          </a>
        </li>
        <li className="nav-item">
          <a href="#" className="nav-link" onClick={() => props.changeCategory('Sports')}>
            Sports
          </a>
        </li>
        <li className="nav-item">
          <a href="#" className="nav-link" onClick={() => props.changeCategory('Hardcover Fiction')}>
            Fiction
          </a>
        </li>
        <li className="nav-item">
          <a href="#" className="nav-link" onClick={() => props.changeCategory('Drama')}>
            Drama
          </a>
        </li>
      </ul>
    </section>
  );
};

BookTypes.propTypes = {
  changeCategory: PropTypes.func.isRequired,
};

export default BookTypes;
