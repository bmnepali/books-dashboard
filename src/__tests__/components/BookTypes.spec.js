import * as React from 'react';

// Load Component
import BookTypes from '../../components/BookTypes';

describe('BookTypes Component', () => {
  describe('rendering testing', () => {
    it('renders BookTypes component without any error', () => {
      shallow(<BookTypes />);
    });

    it('matches snapshot', () => {
      const wrapper = shallow(<BookTypes />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
