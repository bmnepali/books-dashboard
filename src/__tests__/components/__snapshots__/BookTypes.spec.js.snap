// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookTypes Component rendering testing matches snapshot 1`] = `
<section>
  <h3
    className="sidebar-heading"
  >
    Book Types
  </h3>
  <ul
    className="nav flex-column mb-2"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Biography
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Kids
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Sports
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Fiction
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Drama
      </a>
    </li>
  </ul>
</section>
`;
